WooCommerce/Classes/POS/Presentation/ItemListView.swift

```diff
@@ -22,25 +23,28 @@ struct ItemListView: View {
                 ScrollView {
                     ForEach(viewModel.items, id: \.productID) { item in
                         Button(action: {
-                            viewModel.addItemToCart(item)
+                            viewModel.select(item)
                         }, label: {
                             ItemCardView(item: item)
                         })
                     }
                 }
             }
         }
+        .task {
+            await viewModel.populatePointOfSaleItems()
```

Comment:
By moving the `populatePointOfSaleItems()` task from the dashboard to the `ItemListView`, it will also call the remote to re-populate products if we decide to go back from the checkout screen to the product list in order to edit the order (add more button). 

We can perform some additional checks here to bypass this call, or move the call back to the dashboard, but I'll raise with Android what is the actual behaviour that we want for this flow, since can be tackled separately.

Reply:
After testing, maybe we can trigger `populatePointOfSaleItems` only at the beginning of each transaction. If the merchant taps `Add More` to update the cart again, maybe we can keep the item list as it is while the merchant can still pull to refresh the list. An edge case we might want to handle is if a product is no longer in the list after a refresh, while it's in the cart.

Reply:
> maybe we can trigger populatePointOfSaleItems only at the beginning of each transaction. If the merchant taps Add More to update the cart again, maybe we can keep the item list as it is while the merchant can still pull to refresh the list

I agree, this is how it worked previously and makes things faster, but the trade-off is the potential for stale data. Now we have up-to-date data but an additional network request if we go back.

> An edge case we might want to handle is if a product is no longer in the list after a refresh, while it's in the cart.

Ah, good one. I'm not sure what is the expected behaviour but I'd expect it to work the same as with the app if we happen to add an item to an order -> delete the item from the web -> attempt to process the order. I'll log this to add to the testing plan.